common/geth/failover.go

```diff
@@ -0,0 +1,79 @@
+package geth
+
+import (
+	"sync"
+
+	"github.com/Layr-Labs/eigensdk-go/logging"
+)
+
+type FailoverController struct {
+	NumberFault     uint64
+	NumberSuccess   uint64
+	SwitchTrigger   int
+	NumberOfBackups int
+	currentRPCIndex int
+	Logger          logging.Logger
+	mu              *sync.Mutex
+}
+
+func NewFailoverController(numBackup int, switchTrigger int, logger logging.Logger) *FailoverController {
+	return &FailoverController{
+		NumberFault:     0,
+		NumberSuccess:   0,
+		currentRPCIndex: 0,
+		SwitchTrigger:   switchTrigger,
+		NumberOfBackups: numBackup,
+		Logger:          logger,
+		mu:              &sync.Mutex{},
+	}
+}
+
+// To use the Failover controller, one must insert this function
+// after every call that uses RPC.
+// This function attribute the error and update statistics
+func (f *FailoverController) ProcessError(err error) {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	if err == nil {
+		f.NumberSuccess += 1
+		return
+	}
+
+	fault := HandleError(err)
+	if fault == SenderFault {
+		return
+	} else if fault == RPCFault {
+		f.updateRPCFault(err)
+		return
+	} else if fault == Ok {
+		return
+	} else { // TooManyRequest
+		f.updateRPCFault(err)
```

Comment:
The server has no fault if it's too many requests; it's the client to take action